lib/Charts/BarChart.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+import d3Sync from './d3Sync';
+
+import BaseChart from './BaseChart';
+
+const barWidth = 6;
+
+class BarChart extends BaseChart {
+
+    render(chart, chartData, renderContext) {
+        const {scales, state, chartTransition} = renderContext;
+        const chartDataSets = state.data.filter(data => data.renderer instanceof BarChart);
+        const offset = (barWidth * chartDataSets.length) / 2 - chartDataSets.indexOf(chartData) * barWidth;
+        const sx = scales.x, sy = scales.y[chartData.units], color = chartData.color || 'white';
+        
+        // preprocess for negative data; rect does not seem to like negative values.
+        // TODO: Maybe a better way to do this is with transforms.
```

Comment:
Kevin always hassles me about leaving TODOs in the code... can you decide which way is better?